client/ayon_core/hosts/maya/api/lib.py

```diff
@@ -1857,7 +1880,16 @@ def get_container_members(container):
         if ref.rsplit(":", 1)[-1].startswith("_UNKNOWN_REF_NODE_"):
             continue
 
-        reference_members = cmds.referenceQuery(ref, nodes=True, dagPath=True)
+        try:
+            reference_members = cmds.referenceQuery(ref,
+                                                    nodes=True,
+                                                    dagPath=True)
```

Comment:
I tried making this fail but it only gave me `RuntimeError` when using wrong flags otherwise it always return empty any thing or `None`.

Reply:
It's not about the arguments being wrong - it's about the reference node being passed along being an 'invalid' reference.

I was able to reproduce it like this.

1. Create a maya reference (e.g. load something)
2. Select the reference node in the outliner (you might need to enable Display > Reference Nodes in outliner)
3. Duplicate the node (press CTRL+D)

The duplicated reference node will not display in the outliner, but it is a valid reference node:
```python
from maya import cmds
print(cmds.ls(type="reference"))
```

This will then also print the duplicated reference.
However, now if we use that reference node in our queries like this:

```python
from maya import cmds
for reference in cmds.ls(type="reference"):
    cmds.referenceQuery(reference, nodes=True)
```

We get an error:
```
# Error: Reference node 'testRN1' is not associated with a reference file.
# # Traceback (most recent call last):
# #   File "<maya console>", line 4, in <module>
# # RuntimeError: Reference node 'testRN1' is not associated with a reference file.
```

This PR solves that by first checking whether the reference node is 'valid' (associated with a reference file).
I believe there are other ways to get reference nodes to this state than just duplicating it. But this makes at least an easy reproducable.